src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -120,8 +120,13 @@ Copyright (c) .NET Foundation. All rights reserved.
           ItemName="_RestoreGraphEntryFiltered" />
     </RemoveDuplicates>
 
+    <PropertyGroup>
+      <SolutionDir Condition="'$(SolutionDir)'=='' or '$(SolutionDir)'=='*Undefined*'"></SolutionDir>
```

Comment:
I wouldn't clear this property if its `*Undefined*` because other logic might be used to its value being set to that if its not set.  You'll just need to bake this value in as a consideration as null or empty.

Reply:
doesn't this configuration **only** affect `RestoreTask`, which expects this to be a bool anyway?